### ISSUE TITLE
Can't compute distributed gradient in 2 computers with GradientTape

### ISSUE THREAD

**Issue by jorgeraulgomez** [tensorflow/tensorflow#36540]
- TensorFlow version 2.0.0

I am training 2 cascading models, model1 and model2, to be used in 2 different computers. Currently I am simulating it in the same computer. If I train the models in the same gradient tape, it trains perfectly, computing the gradients as:

        import tensorflow as tf
        loss_fn = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True)

        with tf.GradientTape(persistent=True) as tape:
            features = model1(x_batch_train)
            logits = model2(features)
            loss_value = loss_fn(y_batch_train, logits)

        # Computing gradients for model 2
        grads2 = tape.gradient(loss_value, model2.trainable_variables)
        # Computing gradients to pass to model 1
        grads_pass = tape.gradient(loss_value, features)
        # Computing gradients for model 1
        grads1 = tape.gradient(features, model1.trainable_variables, output_gradients=grads_pass)

However, if I want to split the training into 2 different computers, I should have a gradient tape in each one of them, but then the results computed are different!

        import tensorflow as tf
        loss_fn = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True)

        with tf.GradientTape() as tape1:
            features = model1(x_batch_train)
        
        with tf.GradientTape(persistent=True) as tape2:
            logits = model2(features)
            loss_value = loss_fn(y_batch_train, logits)

        # Computing gradients for model 2
        grads2 = tape2.gradient(loss_value, model2.trainable_variables)
        # Computing gradients to pass to model 1
        grads_pass = tape2.gradient(loss_value, features)
        # Computing gradients for model 1
        grads1 = tape1.gradient(features, model1.trainable_variables, output_gradients=grads_pass)

It would be very helpful to have a way to be able to compute gradients in different tapes synchronizing them somehow.

Thanks in advance

**Comment by ravikyram** [comment]
@jorgeraulgomez 

Can you please provide us colab link or simple stand alone code with supporting files to reproduce the issue in our environment. It helps us in localizing the issue faster. Thanks!

**Comment by jorgeraulgomez** [comment]
Yes, sure

https://drive.google.com/file/d/1uCGFklj4EoyrlltgXeOGudMsEGtULbXm/view?usp=sharing

**Comment by gowthamkpr** [comment]
@jorgeraulgomez Please take a look at this [doc](https://github.com/tensorflow/examples/blob/master/community/en/docs/deploy/distributed.md) where it explains how to use tensorflow in a distributed mode. Thanks!

**Comment by jorgeraulgomez** [comment]
Thank you for your answer. It is a very interesting document.

However, I don't think it solves my problem. I am currently executing it in the same notebook, so I am not distributing it yet. This will be a problem of my future me, and I will address it, but for now I would need to be able to execute it in the same notebook.

Regards

**Comment by gowthamkpr** [comment]
@jorgeraulgomez If you are planning to work on the distributed mode then you have to follow the Distributed Tensorflow concept.

Anyways I am going to close this issue for now and if you encounter any issues in future, create a new issue. Thanks!
